### DISCUSSION TITLE
`word-break` option for caption

### DISCUSSION THREAD

**Discussion by zeek0x** [ImageMagick/ImageMagick#5440] [Development] [answered]
Hi :)

In ImageMagick, captions can do word-wrapping. The word-wrapping breaks lines at whitespace if the next word would overflow the line, even if the end of the line has not been reached.

This behavior is common in European language, but does not apply to languages that are not separated by whitespace, such as CJK (Chinese, Japanese, and Korean); In CJK, when the end of lines is reached, the sentence (words) is broken up and sent to the next line, and no line broken if a space appears. However, blanks may appear as an expression of the text.

To control this behavior, I propose the `word-break` option.

---

**word-break option**

`-word-break Type`

Type

- `Normal` (default): Line breaks at space positions when words do words overflow the line. 
- `Break`: When a word overflows a line, the word is split and sent to the next line.

---

If `Normal` is specified, same as current behavior.

**Example**

Normal

```
magick -font /app/Noto_Sans_JP/NotoSansJP-Bold.otf -size 200x -pointsize 24 -word-break Normal caption:"こんにちは！　いい天気ですね！　ご機嫌いかがですか？　一緒にビールを飲みませんか？" normal.png
```

![normal](https://user-images.githubusercontent.com/12657833/184500776-e820cca1-9940-4208-b2dc-2c7cd146bb5f.png)

Break

```
magick -font /app/Noto_Sans_JP/NotoSansJP-Bold.otf -size 200x -pointsize 24 -word-break Break caption:"こんにちは！　いい天気ですね！　ご機嫌いかがですか？　一緒にビールを飲みませんか？" break.png
```

![break](https://user-images.githubusercontent.com/12657833/184500828-9f43af40-9dc9-42ac-8fc7-82b7e7e90407.png)

**diff**

I have tried, and the essential changes are a few lines. I can send the PR as is. 

https://github.com/zeek0x/ImageMagick/commit/56e32e07f779c30d4074d4745ea34b561b1206ae

**references**

- https://developer.mozilla.org/en-US/docs/Web/CSS/word-break

**Comment by urban-warrior** [comment, marked as answer]
No PR necessary.  We can apply your patch directly from your commit, perhaps within a few days.

**Comment by zeek0x** [comment]
6cf5918

**Comment by zeek0x** [comment]
It's in  7.1.0-47 release! Yay!
